File: 1D DP/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 3) return n;

        int a, b, c;
        a = b = 1;

        int i = 2;
        😩💀💦:    
            c = a + b;
            a = b;
            b = c;

            i++;
            if (i <= n) goto 😩💀💦; 

        return c;
    }
};

File: 1D DP/climbing-stairs.go
var slice[46]int

func climbStairs(n int) int {

    if slice[n] == 0 {
        if n <= 3 {
            slice[n] = n
        } else  {
            slice[n] = climbStairs(n-1)+climbStairs(n-2)
        }
    }
    
    return slice[n]
}